core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCPayoutCurve.scala

```diff
@@ -179,9 +198,10 @@ sealed trait DLCPayoutCurveComponent {
 object DLCPayoutCurveComponent {
 
   def apply(points: Vector[OutcomePayoutPoint]): DLCPayoutCurveComponent = {
-    require(points.head.isEndpoint && points.last.isEndpoint,
+    require(points.head.isInstanceOf[OutcomePayoutEndpoint] && points.last
+              .isInstanceOf[OutcomePayoutEndpoint],
```

Comment:
I personally find having a boolean function much nicer (and more efficient for Java) for these kinds of uses. I don't see any reason for getting rid of the `def isEndpoint`